### PR TITLE
feat: support watch `rslib.config` to rebuild

### PR DESCRIPTION
**Pull request by 9aoy**
## Summary

image

image

## Related Links

closes: https://github.com/web-infra-dev/rslib/issues/138

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: packages/core/src/cli/build.ts [hunk 1]
```diff
@@ -14,9 +18,31 @@ export async function build(
     },
   });
 
-  await rsbuildInstance.build({
+  const buildInstance = await rsbuildInstance.build({
     watch: options.watch,
   });
 
+  if (options?.watch) {
```
Comment (Timeless0911) [inline packages/core/src/cli/build.ts:25]:
```suggestion
  if (options.watch) {
```

File: packages/core/src/cli/build.ts [hunk 2]
```diff
@@ -14,9 +18,31 @@ export async function build(
...
+
+    if (options.configFilePath) {
+      files.push(options.configFilePath);
+    }
+
+    onBeforeRestartServer(buildInstance.close);
+
+    watchFilesForRestart(files, async () => {
+      const { content: rslibConfig, filePath: configFilePath } =
+        await loadRslibConfig(options);
```
Comment (Timeless0911) [inline packages/core/src/cli/build.ts:36]: Users can use `build` function as js api, `loadRslibConfig` can not be used here.

File: packages/core/src/cli/commands.ts [hunk 3]
```diff
@@ -108,9 +118,18 @@ export function runCli(): void {
     .description('start Rsbuild dev server of Module Federation format')
     .action(async (options: CommonOptions) => {
       try {
-        const rslibConfig = await loadRslibConfig(options);
-        // TODO: support lib option in mf dev server
-        await startMFDevServer(rslibConfig);
+        const mfDev = async () => {
```
Comment (Timeless0911) [inline packages/core/src/cli/commands.ts:121]: cliMfDev

File: packages/core/src/cli/restart.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import path from 'node:path';
+import chokidar from 'chokidar';
```
Comment (chenjiahan) [inline packages/core/src/cli/restart.ts:2]: lazy import

File: packages/core/src/cli/restart.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+): Promise<void> {
+  if (!files.length) {
+    return;
+  }
+
+  const watcher = chokidar.watch(files, {
+    ignoreInitial: true,
+    // If watching fails due to read permissions, the errors will be suppressed silently.
+    ignorePermissionErrors: true,
+    ignored: ['**/node_modules/**', '**/.git/**', '**/.DS_Store/**'],
```
Comment (chenjiahan) [inline packages/core/src/cli/restart.ts:18]: chokidar v4 does not support glob, this will not work

**Pull request merged**